src/tests/JIT/Regression/clr-x64-JIT/Directory.Build.props

```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project>
+  <Import Project="$([MSBuild]::GetPathOfFileAbove(Directory.Build.props, $(MSBuildThisFileDirectory)..))" />
```

Comment:
I believe this is why you're not seeing a merged entrypoint. The version of Directory.Build.props under src/tests defaults BuildAsStandalone to true which is needed for (the majority of) projects that haven't yet been merged. As you can easily see, in the previous merged test wrappers the parent props file is only imported after the BuildAsStandalone is defaulted to false in the script. The conditional setting is required to let you override it using an environment variable in local builds. Once I fixed that locally, my build started seeing the merged wrapper project because the clause <code>**/*.*proj</code> ends up including the merged test wrapper itself causing msbuild to complain about circular dependency. I fixed it using

<pre>
    &lt;MergedWrapperProjectReference Remove="$(MSBuildThisFileFullPath)" />
</pre>

With this delta I'm able to build and run the merged test wrapper even though some tests seem to fail at runtime, I guess that's the next thing to look at. Please also note that, if the tests to be merged include ilproj tests that use the common suffixes to denote build flavors (<code>_d</code>, <code>_do</code>, <code>_r</code>, <code>_ro</code>, none of the 5 tests in your sample uses those), you'll also need the AssemblyName transform clause you can see in Directory.Build.props under JIT/Methodical.

Reply:
Actually I take that back; I realized I ended up rebuilding the tests without the <code>src\tests\build rebuild</code> clause so that I ended up with an obsolete legacy XUnit wrapper. Now I think all five tests are passing for me (on Windows x64 debug). Surprisingly <code>src\tests\run</code> does report some failures that look like as if the legacy XUnit wrapper mechanism was still trying to run some of the merged tests and fail due to their newly removed entrypoints.

I think part of the problem is that this split scenario is not well supported by the scripts involved. XUnit wrappers are generated by collecting <code>**/*.cmd/sh</code> under each 2-level subtree of <code>XUnitTestBinBase</code>, just excluding subtrees representing merged tests that are marked by special files of the form <code>*.MergedTestAssembly</code>. Effectively, if you have both merged and unmerged tests in the tree, the XUnit wrapper generator will detect there are some cmd / sh scripts in the tree (for the unmerged and out-of-process tests) and it generates the legacy wrapper that subsequently fails for out-of-process csproj tests because those have been bereft of the traditional entrypoint <code>Main</code> - as they're building with BuildAsStandalone set to false, the simple wrappers supplying an entrypoint aren't generated for them, /cc @jkoritzinsky to confirm.

Reply:
I have a vague feeling this is probably fixable and likely worth fixing as that would lower the constraints for merging tests supporting a more piecemeal approach but it's too late for me right now to see what exactly the solution should be, at a first glance we might be able to use a combination of the absence of Exe build mode and RequireProcessIsolation as an indirect indication that the intent of the test is to run as merged and produce a variant of the MergedTestAssembly marker file in its output folder that the legacy XUnit wrapper generator would later use to skip the test.

Reply:
Thank you!  I'm glad I shared this after just 5 tests to catch these things.  I will fix the build files as you suggested.  The split scenario doesn't bother me for just these 5 tests as I will convert more, but it may be an issue since JIT/Regression will be multiple merged batches.